### PR TITLE
Fix reference count in DeleteHistoryBranch

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
Fix reference count in DeleteHistoryBranch

**Why?**
Current reference count only counts ancestors of a branch not the branch itself.

**How did you test it?**
Added persistence test

**Potential risks**

**Is hotfix candidate?**
Yes?

### CODE REVIEW

**Comment by yycptt** [comment]
Will move tests to https://github.com/temporalio/temporal/blob/master/common/persistence/tests/history_store.go

File: common/persistence/history_manager.go [hunk 1]
```diff
@@ -171,6 +171,7 @@ func (m *executionManagerImpl) DeleteHistoryBranch(
 			// skip the target branch
 			continue
 		}
+		usedBranches[br.BranchId] = common.LastEventID
```
Comment (yiminc) [inline common/persistence/history_manager.go:174]: won't this prevent anything from been deleted in the tree except the target branch?

Comment (yycptt) [reply common/persistence/history_manager.go:174]: If target branch's ancestor is not reference by other branches, those ancestors will also be deleted. Not sure if this is your question.

**Review by yiminc — APPROVED** [review]

**Review by wxing1292 — APPROVED** [review]

File: common/persistence/history_manager.go (hunk 1 above)
Comment (yycptt) [reply common/persistence/history_manager.go:174]: Synced offline.

**Pull request merged**